src/Estimators/InputSection.cpp

```diff
@@ -158,7 +155,8 @@ void InputSection::check_valid()
             << " has not been assigned\n";
       throw UniformCommunicateError(error.str());
     }
-}
+  this->checkParticularValidity();
```

Comment:
Why is `this->` needed? InputSection is not a template class.

Reply:
and this